### PR TITLE
Marty2

### PR DESCRIPTION
**Pull request by martyworm**
changes include having a very rough turn exchange happen.  cards are rendered to their correct spots. End turn button works

### CODE REVIEW

File: src/main/java/com/martyworm/Player/Player.java [hunk 1]
```diff
@@ -89,5 +104,15 @@ public void setTurn(boolean turn) {
...
+        return mana;
+    }
+
+    public void setMana(int mana) {
+        this.mana = mana;
+    }
 
-}
+    public void subtractMana(int num){
+        this.mana -= num;
```
Comment (EricH-Propser) [inline src/main/java/com/martyworm/Player/Player.java]: potential bug, can go into negative numbers - is that allowed?

File: src/main/java/com/martyworm/cards/Card.java [hunk 2]
```diff
@@ -59,7 +62,7 @@ public void tick() {
 
     public void render(Graphics g){
 
-        if(inHand) {
+        if (inHand || inActive) {
```
Comment (EricH-Propser) [inline src/main/java/com/martyworm/cards/Card.java]: this doesn't make sense.

File: src/main/java/com/martyworm/cards/Card.java [hunk 3]
```diff
@@ -145,6 +149,15 @@ public boolean hoveringOnRedTile(){
         return false;
     }
 
+    public boolean manaCheck(Player player){
+        if((player.getMana() - mana) >= 0){
```
Comment (EricH-Propser) [inline src/main/java/com/martyworm/cards/Card.java]: Naming: doesPlayerHaveEnoughMana()

you can just:
return (player.getMana() - mana >= 0);

File: src/main/java/com/martyworm/cards/MinionCard.java [hunk 4]
```diff
@@ -10,21 +10,86 @@
...
     public void onClick(){
-
     }
 
     @Override
     protected void cast(Tile t){}
 
     @Override
-    public void turnTilesRed(){}
+    protected void turnTilesRed(int playerId){
```
Comment (EricH-Propser) [inline src/main/java/com/martyworm/cards/MinionCard.java]: I'm unclear on the purpose of this method - can you explain?

File: src/main/java/com/martyworm/cards/Card.java (hunk 2 above)
Comment (martyworm) [reply src/main/java/com/martyworm/cards/Card.java]: Basically rendering the cards in their respective positions if they are either in the active set or in the player's hand.  I don't want to render the cards that are in the graveyard or in the deck

File: src/main/java/com/martyworm/Player/Player.java (hunk 1 above)
Comment (martyworm) [reply src/main/java/com/martyworm/Player/Player.java]: good call

File: src/main/java/com/martyworm/cards/Card.java (hunk 3 above)
Comment (martyworm) [reply src/main/java/com/martyworm/cards/Card.java]: True, thanks

File: src/main/java/com/martyworm/cards/MinionCard.java (hunk 4 above)
Comment (martyworm) [reply src/main/java/com/martyworm/cards/MinionCard.java]: When a minion card is selected while it's in the hand, there are certain places on the board available to summon that minion, ie: the red tiles

Comment (EricH-Propser) [reply src/main/java/com/martyworm/cards/MinionCard.java]: Maybe a name like highlightPlacementTiles would make more sense

**Pull request merged**